File: sdk/swift/Sources/TelecomSDK/GraphQLAPI.swift
import Foundation

/// API for GraphQL queries
public class GraphQLAPI {
    private let client: HTTPClient
    
    public init(client: HTTPClient) {
        self.client = client
    }
    
    /// Execute a GraphQL query
    public func execute(query: String, variables: [String: Any]? = nil) async throws -> [String: Any] {
        let request = GraphQLRequest(query: query, variables: variables)
        return try await client.post(path: "/graphql", body: request)
    }
}

File: sdk/kotlin/src/main/kotlin/com/telecom/GraphQLAPI.kt
package com.telecom

/**
 * API for GraphQL queries
 */
class GraphQLAPI(private val client: HTTPClient) {
    
    /**
     * Execute a GraphQL query
     */
    suspend fun execute(query: String, variables: Map<String, Any>? = null): Map<String, Any> {
        val request = GraphQLRequest(query, variables)
        return client.post("/graphql", request)
    }
}
